src/v2/models/application.js

```diff
@@ -878,6 +878,33 @@ export default class ApplicationModel extends GenericModel {
     return _.filter(namespaces, (ns) => !_.get(ns, 'metadata.name', '').startsWith('openshift') && !_.get(ns, 'metadata.name', '').startsWith('open-cluster-management'));
   }
 
+  async getManagedCluster(clusterName) {
+    const managedCluster = await this.kubeConnector.get(
+      `/apis/cluster.open-cluster-management.io/v1/managedclusters/${clusterName}`,
+    ).catch((err) => {
+      logger.error(err);
+      throw err;
+    });
+    let successImportStatus = true;
+    if (managedCluster.status.conditions) {
```

Comment:
You might want to use lodash to get this array just in case those fields are missing. For example, this is the value you get for a non-existent cluster:

```
{
  kind: 'Status',
  apiVersion: 'v1',
  metadata: {},
  status: 'Failure',
  message: 'managedclusters.cluster.open-cluster-management.io "fakecluster" not found',
  reason: 'NotFound',
  details: {
    name: 'fakecluster',
    group: 'cluster.open-cluster-management.io',
    kind: 'managedclusters'
  },
  code: 404
}
```

It does happen to have a status field, so we're good in that case, but I'm wondering if it's possible in some case that it might not.